grr/server/grr_response_server/gui/ui/components/client/client_test.ts

```diff
@@ -172,49 +180,103 @@ describe('Client Component', () => {
     labelsChipList[0].remove();
-    removedLabelsSubject.error(new Error('Some error'));
+    removedLabelsSubject.next('testlabel');
     fixture.detectChanges();
 
-    const snackbarDiv = document.querySelector('snack-bar-container');
-    expect(snackbarDiv).not.toBeTruthy();
+    expect(facade.addClientLabel).not.toHaveBeenCalled();
+
+    const snackbarDivButton =
+        document.querySelector('div.mat-simple-snackbar-action button');
+    snackbarDivButton!.dispatchEvent(new MouseEvent('click'));
+    fixture.detectChanges();
+    fixture.whenRenderingDone().then(() => {
+      expect(facade.addClientLabel).toHaveBeenCalledWith('testlabel');
+    });
   });
 
-  it('snackbar action undoes a removal of client label', () => {
+  it('allows viewing of client details', () => {
     // Prevent warnings from 404-ing API requests.
     spyOn(facade, 'selectClient');
 
-    const clientSubject = new Subject<Client>();
-    const removedLabelsSubject = new Subject<string>();
-    Object.defineProperty(
-        facade, 'selectedClient$', {get: () => clientSubject});
-    Object.defineProperty(
-        facade, 'lastRemovedClientLabel$', {get: () => removedLabelsSubject});
-    spyOn(facade, 'addClientLabel');
+    const subject = new Subject<Client>();
+    Object.defineProperty(facade, 'selectedClient$', {get: () => subject});
+    spyOn(facade, 'removeClientLabel');
 
     const fixture = TestBed.createComponent(ClientComponent);
     fixture.detectChanges();  // Ensure ngOnInit hook completes.
 
     paramsSubject.next(new Map(Object.entries({id: 'C.1234'})));
-    clientSubject.next(newClient({
+    subject.next(newClient({
       clientId: 'C.1234',
       labels: [{name: 'testlabel', owner: ''}],
     }));
     fixture.detectChanges();
+    const detailsDrawer =
+        fixture.debugElement.query(By.directive(MatDrawer)).componentInstance;
+    const drawerOpenSpy = spyOn(detailsDrawer, 'open');
 
-    const labelsChipList = fixture.debugElement.query(By.directive(MatChipList))
-                               .componentInstance.chips.toArray() as MatChip[];
-    labelsChipList[0].remove();
-    removedLabelsSubject.next('testlabel');
-    fixture.detectChanges();
-
-    expect(facade.addClientLabel).not.toHaveBeenCalled();
+    const detailsButton =
+        fixture.debugElement.query(By.css('.goto-details')).nativeElement;
+    detailsButton.dispatchEvent(new MouseEvent('click'));
 
-    const snackbarDivButton =
-        document.querySelector('div.mat-simple-snackbar-action button');
-    snackbarDivButton!.dispatchEvent(new MouseEvent('click'));
-    fixture.detectChanges();
-    fixture.whenRenderingDone().then(() => {
-      expect(facade.addClientLabel).toHaveBeenCalledWith('testlabel');
-    });
+    expect(drawerOpenSpy).toHaveBeenCalled();
+    detailsDrawer.close();
   });
+
+  it('correctly updates URL for client details drawer usage', fakeAsync(() => {
```

Comment:
This test is too long. If it breaks, the reader will need to inspect 60 LOC. Please split this test up into smaller tests.